src/python/pants/backend/python/packaging/pyoxidizer/rules.py

```diff
@@ -185,8 +187,38 @@ async def package_pyoxidizer_binary(
     )
 
 
+@rule
+async def run_pyoxidizer_binary(field_set: PyOxidizerFieldSet) -> RunRequest:
+    def is_executable_binary(target_name: str, artifact_relpath: str | None) -> bool:
+        """After packaging, the PyOxidizer plugin will place the executable in a location like this:
+        dist/{project}/{target name}/{platform arch}/{compilation mode}/install/{target name}
+
+        e.g. dist/helloworld/helloworld-bin/x86_64-apple-darwin/debug/install/helloworld-bin.
+
+        PyOxidizer will place associated libraries in {...}/install/lib
+
+        To determine if the artifact we iterate over is the one we want to execute, we check that
+        the file's parent dir is "install" and that the filename matches the field_set.target_name
+        """
+        if not artifact_relpath:
+            return False
+
+        artifact_path = Path(artifact_relpath)
```

Comment:
Better is to use `PurePath`, which has the same interface but no fileystem-operations that would break Pants's caching. Even though you aren't using any of those filesystem-methods, it's still a good practice to consistently use `PurePath`.

Reply:
Good point, force of habit :)

Rarely do I only need to do computational and not-IO stuff